File: examples/python/connection.py
import asyncio

import websockets

SERVER = 'wss://api.deepmm.com'


async def connect():
    # Create a WebSocket connection
    open_timeout = 1
    sleep_time = 0
    while True:
        try:
            print(f"Attempting connection to {SERVER}")
            ws = await websockets.connect(SERVER,
                                          max_size=10 ** 8,
                                          open_timeout=open_timeout,
                                          ping_timeout=None)
            print(f"Successful connection to {SERVER}")
            return ws
        except BaseException:
            print(f"Unsuccessful connection to {SERVER}")
            await asyncio.sleep(sleep_time)
            open_timeout = min(60, open_timeout + 1)
            sleep_time = min(10, sleep_time + 1)

File: examples/nodejs/connection.js
export const connect = (onopen, onmessage) => {
    // Create a WebSocket connection
    const SERVER_LIST = ['wss://deist.deepmm.com', 'wss://hayek.deepmm.com'];
    let serverIndex = Math.floor(Math.random() * SERVER_LIST.length);
    let ws = null;
    let connectionTimeoutTimer = null;
    let openTimeout = 1;
    const makeConnection = () => {
        try {
            ws.close();
        } catch {}
        console.log('Attempting connection to ' + SERVER_LIST[serverIndex]);
        ws = new WebSocket(SERVER_LIST[serverIndex]);
        connectionTimeoutTimer = setTimeout(() => {
            serverIndex = serverIndex == SERVER_LIST.length - 1 ? 0 : serverIndex + 1;
            openTimeout += 1;
            makeConnection();
        }, openTimeout * 1000);
        ws.onopen = () => {
            if (connectionTimeoutTimer !== null) {
                clearTimeout(connectionTimeoutTimer);
                connectionTimeoutTimer = null;
            }
            console.log('Successful connection to ' + SERVER_LIST[serverIndex]);
            onopen(ws);
        }
        ws.onmessage = (event) => {
            onmessage(ws, event);
        }
    }
    makeConnection();
};
